pep-0622.rst

```diff
@@ -280,8 +280,67 @@ section for more details.
 Syntax and Semantics
 ====================
 
-Case clauses
-------------
+Patterns
+--------
+
+The **pattern** is a new syntactical construct, that could be considered a loose
+generalization of assignment targets. The key properties of a pattern are what
+types and shapes of subjects it accepts, what variables it captures and how
+it extracts them from the subject. For example the pattern ``[a, b]`` matches
+only sequences of exactly 2 elements, extracting the first element into ``a``
+and the second one into ``b``.
+
+This PEP defines several types of patterns. These are certainly not the
+only possible ones, so the design decision was made to choose a subset of
+functionality that is useful now but conservative. More patterns can be added
+later as this feature gets more widespread use. See the `rejected ideas`_
+and `deferred ideas`_ sections for more details.
+
+The patterns listed here are described in more detail below, but summarized 
+together in this section for simplicity:
+
+- A **capture pattern** looks like ``x`` and is equivalent to an identical
+  assignment target: it always matches and binds the variable
+  with the given name.
+- A **sequence pattern** looks like ``[a, *rest, b]`` and is similar to 
+  a list unpacking. An important difference is that the elements nested
+  within it can be any kind of patterns patterns, not just names or sequences.
+  It matches only sequences of appropriate length, as long as all the sub-patterns
+  also match. It makes all the bindings of its sub-patterns.
+- The **wildcard pattern** is a single underscore: ``_``. It always matches,
+  but does not capture any variable (which prevents interference with other
+  uses that you could have for ``_`` and allows some optimizations). 
```

Comment:
This is a good addition, and now I realize the wildcard pattern isn't listed separately later -- it's hidden in the capture pattern. Also the order patterns are presented here is different from how they are presented later. I'd like this to be consistent.

Reply:
Also maybe wildcard should immediately follow capture? And capture should mention `_` is an exception.

Reply:
I choose this order here to highlight the capture over the constant/literals. I understand that the order later on is design to build the syntax bottom up.